src/Parsers/ParserCreateQuery.h

```diff
@@ -185,15 +186,23 @@ bool IParserColumnDeclaration<NameParser>::parseImpl(Pos & pos, ASTPtr & node, E
     }
 
     Pos pos_before_specifier = pos;
-    if (s_default.ignore(pos, expected) || s_materialized.ignore(pos, expected) ||
-        s_ephemeral.ignore(pos, expected) || s_alias.ignore(pos, expected))
+    if (s_default.ignore(pos, expected) || s_materialized.ignore(pos, expected) || s_alias.ignore(pos, expected))
     {
         default_specifier = Poco::toUpper(std::string{pos_before_specifier->begin, pos_before_specifier->end});
 
         /// should be followed by an expression
         if (!expr_parser.parse(pos, default_expression, expected))
             return false;
     }
+    else if (s_ephemeral.ignore(pos, expected))
+    {
+        if (s_ephemeral.ignore(pos, expected))
```

Comment:
this doesn't seem right

Reply:
yes, looks very strange, especially I have tested it and it worked :)
will check

Reply:
should be fixed now